lib/matplotlib/tests/test_pyplot.py

```diff
@@ -381,3 +383,15 @@ def test_pylab_integration():
         )),
         timeout=60,
     )
+
+
+def test_doc_pyplot_summary():
+    """Test that pyplot_summary lists all the plot functions."""
+    pyplot_docs = Path(__file__).parent / '../../../doc/api/pyplot_summary.rst'
+    if not pyplot_docs.exists():
+        pytest.skip("Documentation sources not available")
+
+    lines = pyplot_docs.read_text()
+    m = re.search(r':nosignatures:\n\n(.*?)\n\n', lines, re.DOTALL)
+    functions = [line.strip() for line in m.group(1).split('\n')]
+    assert functions == plt.get_plot_commands()
```

Comment:
this is going to be pretty mysterious for most folks when they add a pyplot method to axes, and it causes this to fail.  maybe a more verbose failure message?   It seems if we make whole function docstrings with boilerplate, we could also make the list of functions in `pyplot_summary` as well?

Reply:
I hope that most folks will not be adding pyplot functions. That interface should not grow except for very rare cases such as `stairs()`, but sure I will add a more instructive error message.

> It seems if we make whole function docstrings with boilerplate, we could also make the list of functions in `pyplot_summary` as well?

We could along with boilerplate. However, that list is very stable, changes probably less than once a year. I've saved the effort to write a complete generator. Only checking that it's up-to-date is much simpler. And similar to boilerplate for the code we anyway have to check whether it's updated. If not up-to-date, the fix is very easy. Telling the user: "add your_func to the list of plotting commands in `pyplot_summary.rst`" is the same effort and complexity for the user as "run this boilerplate script".

Reply:
Sure, so maybe just a more verbose message here....  It does mean people have to change two things, but agree that the list should rarely change so its not an undue burden...

Reply:
I've added a more instructive message.